### ISSUE TITLE
Clarify the meaning of properties which have default value

### ISSUE THREAD

**Issue by lllyfeng** [netmod-wg/yang-next#115]
Some statements have default value, such as config/mandatory/min-elements/max-elements/status etc. if these statements are not occur under parent nodes, and should we think the parent nodes have these properties actually?
Pyang consider ‘config’ statement always exists. But it don’t  consider other properties with default value always exists.

Libyang doesn’t consider any properties with default value always exists.

If you want to change the value of those properties by using deviation,  you will find you can’t get successful results from pyang and libyang simultaneously.
For example:

```
leaf foo {
   type int8;
}
```

If you want to change config property to false.
If you define the deviation like below,

```
deviation /foo {
    deviate replace {
          config false;
    }
}
```

You will get an error from libyang for the config statement of target not exists.
If you define the deviation like below,

```
deviation /foo {
    deviate add{
          config false;
    }
}
```

You also get an error from pyang for the config statement of target already exists.

It cause confusion, and need clarify in YANG NEXT.

**Comment by jovatn** [comment]
I agree. Today it seems impossible to form such deviate statement, and simultaneously comply with tools using libyang and pyang respectively. See [pyang](https://github.com/mbj4668/pyang/issues/267) and [libyang](https://github.com/CESNET/libyang/pull/1142) discussion threads.

**Comment by abierman** [comment]
The deviation statements are way too picky.
IMO they need to be much more lenient.
Agree with this ticket that the statements with defaults should be handled consistently.

IMO, deviate-stmt should be more lenient:
- add : add if multiple allowed, or error if already set (no change)
- replace: add if not set or replace if match found
- delete: delete if found, not an error if not found

Must Do: complexity: low, bc: medium, importance: high

**Comment by BalazsLengyel** [comment]
Importance high.
One more reason because beginner designers always have problems with this issue.
deviate replace should work even if the statement does not yet exist, just as in an <edit-config>
